docs/developer-guide.md

```diff
@@ -0,0 +1,91 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Build Structure of spark-bench](#build-structure-of-spark-bench)
+- [Adding a New Data Generator](#adding-a-new-data-generator)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+#spark-bench Developer's Guide
+
+Contributions welcome!
+
+## Build Structure of spark-bench
+
+spark-bench is a multi-project SBT build. The build is mainly defined in [build.sbt](../build.sbt) and dependencies
+are defined in [Dependencies.scala](../project/Dependecies.scala).
+
+## Adding a New Data Generator
+
+All the data generators live in the data generator project. Let's add a new data generator called FooGenerator.
+
+The FooGenerator is going to output however many rows and columns of the string you specify, over and over again.
+Pretty useful, right?
+
+We probably want a default string for the FooGenerator to output. Let's make a defaults file. This should
```

Comment:
I'm going to leave this for now because some upcoming changes that Matthew is working on are probably going to change how this all works, but I do agree that having one Defaults file for everything is probs a bad call.